plugins/worker/server/__init__.py

```diff
@@ -91,6 +94,32 @@ def schedule(event):
         })
 
 
+def cancel(event):
+    """
+    This is bound to the "jobs.cancel" event, and will be triggered any time
+    a job is canceled. This handler will process any job that has the
+    handler field set to "worker_handler".
+    """
+    job = event.info
+    if job['handler'] == 'worker_handler':
+        # Stop event propagation and prevent default, we are using a custom state
+        event.stopPropagation().preventDefault()
+
+        celeryTaskId = job.get('celeryTaskId')
+
+        if celeryTaskId is None:
+            msg = ("Unable to cancel Celery task. Job '%s' doesn't have a Celery task id."
+                   % job['_id'])
+            logger.warn(msg)
+            return
+
+        # Send the task to celery
+        asyncResult = AsyncResult(celeryTaskId)
+        asyncResult.revoke()
```

Comment:
@kotfic I'm not appose to added a terminate and signal parameter to this endpoint, which I think is all that would required for your use case? With the appropriate warnings of course :smile: